_posts/2022-08-14-data-pipelines-production-ready-great-expectations.md

```diff
@@ -0,0 +1,102 @@
+excerpt_separator: <!--more-->
+---
+
+Great Expectations is an open source Python library that allows you to write
+detailed tests called
+[expectations](https://docs.greatexpectations.io/docs/terms/expectation/)
+against your data. Users write these expectations to run validations against the
+data as it enters your system. These expectations are expressed as methods in
+Python and stored in JSON and YAML files. One great advantage of expectations is
+the human readable documentation that results from these tests. As you role out
+different versions of the code, you get alerted to any unexpected changes and 
+have version specific generated documentation for what changed. In this blog, 
+we'll show you how to write expectations on tables in Trino!
+
+![](/assets/blog/data-pipelines-production-ready-great-expectations/trino-ge.png)
+
+
+<!--more-->
+
+## The need for data quality
+
+Managing data pipelines is not for the faint of heart. Nodes fail, you run
+out of memory, bursty traffic causes abnormal behavior, and that’s just the tip
+of the Apache Iceberg 😉. Lots of Trino community members build sophisticated
+data pipelines and data applications using Trino. This received a turbo-boost 
+when Trino added a 
+[fault tolerance]({% post_url 2022-05-05-tardigrade-launch %}) mode to safeguard
+against some of these failures when executing long-running and 
+resource-intensive queries. 
+
+Aside from all the infrastructure problems that concern data teams, another
+category of problems that has been the silent killer for quite some time is data
+quality. Faulty data comes in, which can either cause data pipelines to fail, or
+possibly go unnoticed and cause inaccurate downstream reporting. Tacit knowledge
+is scattered among domain experts, technical experts, and the code and data
+itself. Maintenance becomes time-consuming and expensive. Documentation gets out
+of date and unreliable. This is why using data quality checks using libraries 
+like [Great Expectations](https://greatexpectations.io/) is so important when
+writing ETL applications.
+
+## Improve data quality in Trino with Great Expectations
+
+As data quality becomes more in the forefront of the Trino community, the Great
+Expectations and Trino communities have partnered to do some events together:
+* [Trino meetup to discuss Great Expectations](https://www.youtube.com/watch?v=pcqAOq3O3Ts&list=PLFnr63che7wZij92ynF_egatbsrH7by7T&index=3)
+* [Great Expectations meetup to discuss Trino](https://www.youtube.com/watch?v=4SieRmibb0U).
+* [Superconductive](https://superconductive.ai/) joined this year’s mini Trino 
+Summit event 
+[Cinco de Trino](https://www.youtube.com/watch?v=kfJ63DNbAuI&list=PLFnr63che7wYDHjUsmp43THLmAlqPDHlM)
+to showcase using 
+[managed solutions for Great Expectations and Trino](https://www.youtube.com/watch?v=9HE6LawCHP8&list=PLFnr63che7wYDHjUsmp43THLmAlqPDHlM&index=7).
+
+Today, we’re showcasing a demo that showcases a scenario with Trino running as
```

Comment:
"showcasing a demo that showcases" sounds very redundant, maybe "walking through a demo that showcases" or something along those lines just so it's not repetitive.

Reply:
lolz